tests/profiler/tensorflow2/test_native_tf2_profiler.py

```diff
@@ -0,0 +1,454 @@
+
+    detailed_profiler_config = "{"
+    if start_step_2:
+        detailed_profiler_config += _convert_key_and_value("StartStep", start_step_2)
+    if num_steps_2:
+        detailed_profiler_config += _convert_key_and_value("NumSteps", num_steps_2)
+    if start_time:
+        detailed_profiler_config += _convert_key_and_value("StartTimeInSecSinceEpoch", start_time)
+    if duration:
+        detailed_profiler_config += _convert_key_and_value("DurationInSeconds", duration)
+    detailed_profiler_config += "}"
+
+    full_config = {
+        "ProfilingParameters": {
+            "ProfilerEnabled": True,
+            "LocalPath": "/tmp/test",
+            "PythonProfilingConfig": python_profiler_config,
+            "DetailedProfilingConfig": detailed_profiler_config,
+        }
+    }
+
+    with open(profiler_config_path, "w") as f:
+        json.dump(full_config, f)
+
+    profiler_config_parser = ProfilerConfigParser()
+    assert profiler_config_parser.profiling_enabled
+
+    return profiler_config_parser
+
+
+def set_up_profiling(profilerconfig):
+    profiler_config_parser = profilerconfig
+    python_profiler = None
+    if profiler_config_parser.profiling_enabled:
+        config = profiler_config_parser.config
+        if config.python_profiling_config.is_enabled():
+            python_profiler = PythonProfiler.get_python_profiler(config, "tensorflow")
+            python_profiler.start_profiling(StepPhase.START)
+            atexit.register(python_profiler.stop_profiling, StepPhase.END)
+    return profiler_config_parser, python_profiler
+
+
+def create_model():
+    model = tf.keras.models.Sequential(
+        [
+            # WA for TF issue https://github.com/tensorflow/tensorflow/issues/36279
+            tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return model
+
+
+def prepare_dataset():
+    mnist = tf.keras.datasets.mnist
+    (x_train, y_train), _ = mnist.load_data()
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (tf.cast(x_train[..., tf.newaxis] / 255, tf.float32), tf.cast(y_train, tf.int64))
+    )
+    dataset = dataset.shuffle(1000).batch(64)
+    return dataset
+
+
+def helper_native_tf2_gradtape(
+    hook, debugger=False, python_profiler=None, start_step=None, end_step=None
+):
+    def get_grads(images, labels):
+        return model(images, training=True)
+
+    @tf.function
+    def train_step(images, labels):
+        return tf.reduce_mean(get_grads(images, labels))
+
+    dataset = prepare_dataset()
+    model = create_model()
+    opt = tf.keras.optimizers.Adam()
+    hook.wrap_optimizer(opt)
+
+    current_step = 0
+    n_epochs = 1
+    for epoch in range(n_epochs):
+        for data, labels in dataset:
+            labels = tf.one_hot(labels, depth=10)
+            hook.profiling_start_batch()
```

Comment:
nit: can you make this the first line in the second for loop? as in, this should come before the initialization of `labels` above (we want the profiling function to be the first line so that such an initialization is captured in profiling)